lib/tts_isolate.dart

```diff
@@ -83,6 +83,16 @@ void vocalizeMessage(Map<String, dynamic>? message) {
     return;
   }
 
+  //check if message model is StreamStateEventModel
+  if (message.containsKey('isOnline')) {
+    bool isOnline = message['isOnline'] as bool;
+
+    if (!isOnline) {
+      TextToSpeechPlugin.stopSpeaking();
```

Comment:
This is on the correct thread now, however it won't disable the icon either. The icon will still show that TTS is enabled. Think about what manages the state of TTS being enabled/disabled? This isolate listens on that state, we need to adjust the button to also listen on that state.